### PR TITLE
Tell Windows whether to check cert revocation based on the SSLContext's verify_flags

### PR DESCRIPTION
**Pull request by davisagli**
refs #42

### CODE REVIEW

**Comment by davisagli** [comment]
(not sure yet why the tests fail)

File: src/truststore/_api.py [hunk 1]
```diff
@@ -85,6 +85,12 @@ def load_verify_locations(
     def __getattr__(self, name: str) -> Any:
         return getattr(self._ctx, name)
 
+    def __setattr__(self, name: str, value: Any) -> None:
```
Comment (sethmlarson) [inline src/truststore/_api.py:88]: Is there any situation where we don't want to forward `__setattr__` to the inner `_ctx` aside from `verify_mode`? Almost feel like we should reverse our strategy in `__setattr__` here to default to `setattr(self._ctx, name, value)`.

Comment (davisagli) [reply src/truststore/_api.py:88]: I tried it that way first but it caused infinite recursion when setting `_ctx` itself, and when I wrote the conditional to only exclude `_ctx` there were still a few tests failing. Worth some more thought.

**Comment by sethmlarson** [comment]
@davisagli Are you still good with this approach? I couldn't think of a better way to implement this at the moment. We may want to future-proof our `SSLContext` class to only allow a subset of attributes/methods so we're not falsely "advertising" support for new features added to `ssl.SSLContext` in the future?

**Comment by davisagli** [comment]
> We may want to future-proof our SSLContext class to only allow a subset of attributes/methods so we're not falsely "advertising" support for new features added to ssl.SSLContext in the future?

Ah, this might be a good reason for sticking with the wrapper and not the approach I just showed in #51

**Comment by davisagli** [comment]
@sethmlarson There's more to do to make sure we're passing flags sensibly and to make sure the handling of the cert revocation flags makes sense on Mac, but I think we can go ahead and merge this as a fix for Windows.

File: src/truststore/_windows.py [hunk 2]
```diff
@@ -330,6 +330,13 @@ def _verify_peercerts_impl(
         chain_params.cbSize = sizeof(chain_params)
         pChainPara = pointer(chain_params)
 
+        if ssl_context.verify_flags & ssl.VERIFY_CRL_CHECK_CHAIN:
```
Comment (sethmlarson) [inline src/truststore/_windows.py:333]: Should we be starting with `chain_flags = 0` and then `|=` each Windows flag value onto `chain_flags` if the given `ssl.VERIFY_CRL_*` flag value is set?

In it's current state it looks like we can only either check the chain or the end cert, not both.

Comment (davisagli) [reply src/truststore/_windows.py:333]: The revocation check flags for the wincrypt APIs are mutually exclusive; we can set CERT_CHAIN_REVOCATION_CHECK_CHAIN or CERT_CHAIN_REVOCATION_CHECK_END_CERT but not both.

For both OpenSSL's flags and wincrypt's flags, it appears that CHECK_CHAIN also includes checking the leaf cert, so I think the way I've written this preserves the semantics without sending both flags to wincrypt at once.

Comment (sethmlarson) [reply src/truststore/_windows.py:333]: Got it, that makes sense! Thanks for clarifying.

**Pull request merged**